tests/integration/azure_blob/test_azure_blob.py

```diff
@@ -19,15 +19,65 @@
 blob_dir = "test_mlrun_azure_blob"
 blob_file = f"file_{random.randint(0, 1000)}.txt"
 
-
-def azure_connection_configured():
-    return config["env"].get("AZURE_STORAGE_CONNECTION_STRING") is not None
+AUTH_VARS = [
+    "AZURE_STORAGE_CONNECTION_STRING",
+    "AZURE_STORAGE_ACCOUNT_NAME",
+    "AZURE_STORAGE_ACCOUNT_KEY",
+    "AZURE_STORAGE_SAS_TOKEN",
+    "AZURE_STORAGE_CLIENT_ID",
+    "AZURE_STORAGE_CLIENT_SECRET",
+    "AZURE_STORAGE_TENANT_ID",
+]
+
+AUTH_METHODS = [
+    "conn_str",
+    "sas_token",
+    "account_key",
+    "spn",
+]
```

Comment:
In the `azure_connection_configured` function you're testing for all auth methods parameters to be configured. I think this is a bit of an overkill - if I just want to test a single method, this will not run the test unless I configure all the auth methods - SAS token, secrets, connection string etc. 
I propose to configure it differently, something like this:

```suggestion
AUTH_METHODS_AND_REQUIRED_PARAMS = {
    "conn_str": [
        "AZURE_STORAGE_CONNECTION_STRING"
    ],
    "sas_token": [
        "AZURE_STORAGE_ACCOUNT_NAME", "AZURE_STORAGE_SAS_TOKEN"
    ],
    "account_key": [
        "AZURE_STORAGE_ACCOUNT_NAME", "AZURE_STORAGE_ACCOUNT_KEY"
    ],
    "spn": [
        "AZURE_STORAGE_ACCOUNT_NAME",
        "AZURE_STORAGE_CLIENT_ID",
        "AZURE_STORAGE_CLIENT_SECRET",
        "AZURE_STORAGE_TENANT_ID",
    ]
}
```

Then in the `azure_connection_configured` method you can just verify that a single auth method got all of the parameters needed for it. 
In the `prepare_env` implementation you just need to iterate over the parameters in the method selected and assign the OS environment from the config (assuming the params are configured, of course). If params are not configured for a given auth method, the `prepare_env` can return an indication to that, and the test will skip running that method.

Reply:
Coming to think of it - if doing this then can completely remove the `@pytest.mark.skipif` in the beginning of the tests - simply iterate over all auth methods, and only perform the test if params are configured for it. If none of the methods are configured, the test will succeed by default.